cpp/tests/community/k_truss_test.cu

```diff
@@ -0,0 +1,408 @@
+            h_dst.resize(thrust::distance(edge_first, edge_last));
+
+            n_dropped += 1;
+          }
+        }
+      }
+    }
+
+    // convert back to device_vector
+    rmm::device_uvector<vertex_t> d_reference_src(h_src.size(), handle.get_stream());
+
+    raft::update_device(d_reference_src.data(), h_src.data(), h_src.size(), handle.get_stream());
+
+    rmm::device_uvector<vertex_t> d_reference_dst(h_src.size(), handle.get_stream());
+
+    raft::update_device(d_reference_dst.data(), h_dst.data(), h_dst.size(), handle.get_stream());
+
+    auto d_reference_wgt =
+      std::make_optional<rmm::device_uvector<weight_t>>(h_src.size(), handle.get_stream());
+
+    if (edge_weight_view) {
+      raft::update_device(
+        (*d_reference_wgt).data(), h_wgt.data(), h_wgt.size(), handle.get_stream());
+    }
+
+    return std::make_tuple(std::move(d_reference_src),
+                           std::move(d_reference_dst),
+                           edge_weight_view ? std::move(d_reference_wgt) : std::nullopt);
+  }
+
+  template <typename vertex_t, typename edge_t, typename weight_t>
+  void run_current_test(std::tuple<KTruss_Usecase const&, input_usecase_t const&> const& param)
+  {
+    constexpr bool renumber = false;
+
+    auto [k_truss_usecase, input_usecase] = param;
+
+    raft::handle_t handle{};
+
+    HighResTimer hr_timer{};
+
+    if (cugraph::test::g_perf) {
+      RAFT_CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+      hr_timer.start("MG Construct graph");
+    }
+
+    // NX k_truss is not implemented for graph with self loop and isolated
+    // vertices therefore dropped them especially for rmat generated graphs
+    auto [graph, edge_weights, d_renumber_map_labels] =
+      cugraph::test::construct_graph<vertex_t, edge_t, weight_t, false, false>(
+        handle, input_usecase, k_truss_usecase.test_weighted_, renumber, true, true);
+
+    if (cugraph::test::g_perf) {
+      RAFT_CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+      hr_timer.stop();
+      hr_timer.display_and_clear(std::cout);
+    }
+
+    auto graph_view = graph.view();
+
+    auto edge_weight_view =
+      edge_weights ? std::make_optional((*edge_weights).view()) : std::nullopt;
+
+    if (cugraph::test::g_perf) {
+      RAFT_CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+      hr_timer.start("K-truss");
+    }
+
+    auto [d_cugraph_src, d_cugraph_dst, d_cugraph_wgt] =
+      cugraph::k_truss<vertex_t, edge_t, weight_t, false>(
+        handle, graph_view, edge_weight_view, k_truss_usecase.k, false);
+
+    if (cugraph::test::g_perf) {
+      RAFT_CUDA_TRY(cudaDeviceSynchronize());  // for consistent performance measurement
+      hr_timer.stop();
+      hr_timer.display_and_clear(std::cout);
+    }
+
+    if (k_truss_usecase.check_correctness) {
+      auto [d_reference_src, d_reference_dst, d_reference_wgt] =
```

Comment:
See https://github.com/rapidsai/cugraph/blob/branch-24.04/cpp/tests/cores/core_number_test.cpp#L283

Reply:
the reference k-truss is now leveraging the CSR format